src/types/embedded.ts

```diff
@@ -0,0 +1,29 @@
+import deserialize from "../core/deserialize";
+import serialize from "../core/serialize";
+import { SKIP } from "../constants";
+import custom from "./custom";
+import { ClazzOrModelSchema } from "../api/types";
+
+/**
+ * This allows to embed the property values in the resulting json output
+ * and vice-versa.
+ *
+ * @param type {ClazzOrModelSchema<T>} Some class or model schema.
+ */
```

Comment:
Please add an example of the usage here. See the other built in types for how to format it so that it shows up properly in the README.